### PR TITLE
Feat

### LINKED ISSUE #15: [Feature] Share Question Button
### Description

**Add one button Where the Link is Copy when the Button is clicked and That Link Redirect that particular Question** 

### Screenshots

image

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/Nishitbaria/Dev-Exchange/blob/main/CONTRIBUTING.md)
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by thedemonsid**
## Fixes Issue
Closes #15 
## Changes proposed

- [x] Added the Copy button which Copies the Question URL to Clipboard of user
- [x] Made a separate CoppyButton Component so that the question/[id] page does not becomes client component.

## Screenshots
Screenshot from 2024-05-12 15-36-09

## Note to reviewers

- [x] Added additional functionalty to copy the editor content.
 If any problem let me know

### CODE REVIEW

**Comment by Nishitbaria** [comment]
> 
> ## Fixes Issue
> Closes #15 
> ## Changes proposed
> 
> - [x] Added the Copy button which Copies the Question URL to Clipboard of user
> - [x] Made a separate CoppyButton Component so that the question/[id] page does not becomes client component.
> 
> ## Screenshots
> Screenshot from 2024-05-12 15-36-09
> 
> ## Note to reviewers
> 
> - [x] Added additional functionalty to copy the editor content.
>  If any problem let me know
> 

Instead of copy button add copy icon which looks better

**Comment by thedemonsid** [comment]
@Nishitbaria Having difficulty with passing all checks , any suggestions ??

**Comment by Nishitbaria** [comment]
> @Nishitbaria Having difficulty with passing all checks , any suggestions ??

on which pass ?

**Comment by thedemonsid** [comment]
4 workflows awaiting approval
image

**Comment by thedemonsid** [comment]
@Nishitbaria done with checks

**Comment by Nishitbaria** [comment]
> @Nishitbaria done with checks

I don't want copy button like this : 

WhatsApp Image 2024-05-12 at 16 18 09_95c6bf01


I want copy button like this  : 
image

**Comment by thedemonsid** [comment]
Use react luci icon and only add copy button not others

**Comment by Nishitbaria** [comment]
> Use react luci icon and only add copy button not others 

Yes

**Comment by thedemonsid** [comment]
@Nishitbaria done with the changes .

**Comment by Nishitbaria** [comment]
> @Nishitbaria done with the changes .

WhatsApp Image 2024-05-12 at 19 37 34_11a96b19

please fix this issue

**Comment by thedemonsid** [comment]
okay

**Comment by thedemonsid** [comment]
image
 @Nishitbaria done with changes

File: components/shared/ParseHTML.tsx [hunk 1]
```diff
@@ -25,15 +25,22 @@ import "prismjs/components/prism-sql";
 import "prismjs/components/prism-mongodb";
 import "prismjs/plugins/line-numbers/prism-line-numbers.js";
 import "prismjs/plugins/line-numbers/prism-line-numbers.css";
-
 interface Props {
   data: string;
 }
 
 export default function ParseHTML({ data }: Props) {
+  const [copied, setCopied] = useState(false);
```
Comment (Nishitbaria) [inline components/shared/ParseHTML.tsx]: In Versel It gives Error Fix it You have defined the copied state but where are you passing this? 


```
<html>
<body>
<!--StartFragment-->
Error: 'copied' is assigned a value but never used.  no-unused-vars
--
21:17:13.324 | 34:9  Error: 'handleCopy' is assigned a value but never used.  no-unused-vars

<!--EndFragment-->
</body>
</html>
```

Comment (Nishitbaria) [reply components/shared/ParseHTML.tsx]: Hey @thedemonsid  fix the deployment issue : 


image

**Comment by thedemonsid** [comment]
@Nishitbaria  Removed unwanted code please check the deployment

**Comment by thedemonsid** [comment]
@Nishitbaria by mstake unassigned myself , can you reassign me ??

**Comment by Nishitbaria** [comment]
I tell you to add a copy button at the left side of the upvote button @thedemonsid

**Comment by thedemonsid** [comment]
image
like this ??

**Comment by thedemonsid** [comment]
@Nishitbaria can you approve the checking process ??
Also can you reassign the issue related to it to me as i have accidentely removed myself from the issue

**Comment by thedemonsid** [comment]
@Nishitbaria can you approve checking process ??
Done with code

**Comment by Nishitbaria** [comment]
> @Nishitbaria can you approve checking process ?? Done with code

@thedemonsid  I want this type of Share button 

https://github.com/Nishitbaria/Dev-Exchange/assets/85815172/e7ad080b-3637-4358-8f2f-3c0e130677d9


Take reference of this https://ui.shadcn.com/docs/components/dialog

**Pull request closed without merging**